src/psyclone/transformations.py

```diff
@@ -170,6 +177,25 @@ def _validate(self, node_list):
                     "transformation to the IfBlock node instead.".
                     format(self.name))
 
+        # The checks below this point only apply to the NEMO API
+        node = node_list[0]
+        if not isinstance(node.root, NemoInvokeSchedule):
```

Comment:
I agree but short of creating a NemoRegionTrans and then implementing NEMO-specific versions of the various OpenMP/OpenACC transformations I think it's hard to avoid.

Reply:
OK.